File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private long BFS(TreeNode root,int k){
        int l,i;
        PriorityQueue<Long> heap=new PriorityQueue<>();
        Queue<TreeNode> q=new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()){
            l=q.size();
            long sum=0;
            for(i=0;i<l;i++){
                TreeNode r=q.poll();
                sum+=r.val;
                if(r.left!=null){q.add(r.left);}
                if(r.right!=null){q.add(r.right);}
            }
            heap.add(-sum);
        }
        for(i=0;i<k-1;i++){
            if(!heap.isEmpty()){
                heap.poll();
            }
        }
        if(!heap.isEmpty()){
            return -heap.poll();
        }
        else{
            return -1;
        }
    }
    public long kthLargestLevelSum(TreeNode root, int k) {
        return BFS(root,k);
    }
}

File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:
        def BFS(root,k):
            queue=deque([root])
            heap=[]
            while queue:
                l=len(queue)
                sum=0
                for i in range(l):
                    #print('heap',list(heap))
                    #print(list(queue))
                    r=queue.popleft()
                    #print('val',r.val)
                    sum+=r.val
                    if r.left is not None:
                        #print('r.left',r.left.val,'----',r.val)
                        queue.append(r.left)
                    if r.right is not None:
                        #print('r.right',r.right.val,'----',r.val)
                        queue.append(r.right)
                #print(sum)
                heapq.heappush(heap,-sum)
            for i in range(k-1):
                if heap:
                    heapq.heappop(heap)
            if heap:
                return -heapq.heappop(heap)
            else:
                return -1
        return BFS(root,k)
        